### ISSUE TITLE
Add support for passive scanning

### ISSUE THREAD

**Issue by umgefahren** [esp-rs/embedded-svc#57]
My application requires passive wifi scanning wich is not possible through the trait. My suggestion would be to add some config for the wifi scanning. I also consider it suboptimal that a Vec is returned from the scan method. Why not an iterator?

**Comment by ivmarkov** [comment]
> My application requires passive wifi scanning wich is not possible through the trait. My suggestion would be to add some config for the wifi scanning. 

This project is largely driven by community contributions s, so the fastest way to get what you want is to work on a PR.

> I also consider it suboptimal that a Vec is returned from the scan method. Why not an iterator?

Because the underlying ESP IDF driver does not allow that.

**Comment by ivmarkov** [comment]
By the way - you've opened this issue in the wrong g project. Please move it to `esp-idf-svc` and close this one.

**Comment by ivmarkov** [comment]
In fact, passive scanning [is already supported in `esp-idf-svc`](https://github.com/esp-rs/esp-idf-svc/blob/master/src/wifi.rs#L796). It is only not exposed in the `Wifi` trait, that's all.

**Comment by umgefahren** [comment]
But in order to be exposed in the WIFI trait there needs to be a trait method, right?

I will draft up a PR for that as soon as possible.

**Comment by ivmarkov** [comment]
Why do you want to go for this via the `Wifi` trait? Are you trying to somehow keep your code abstract, so that it works on something else besides Espressif chips?

If not, then this effort is not worth it.

**Comment by umgefahren** [comment]
Yes and I generally prefer the high level abstractions. I thought it might be a good idea since it's supported by a lot of wifi drivers. It would also enable to pass detailed options on how to perform the scanning.

**Comment by ivmarkov** [comment]
Well, fine then. Please work on a PR, and then we can discuss once we have your suggested changes.

In any case, getting your feet wet with the `esp-idf-*` echosystem and trying passive scan by directly calling the `EspWifi` structure is highly advisable before trying a PR.
